clang/include/clang/Basic/Attr.td

```diff
@@ -1816,7 +1841,7 @@ def SYCLAddIRAttrCommonMembers : SYCLAddIRAttrMemberCodeHolder<[{
          ValueQType->getArrayElementTypeNoTypeQual()
                    ->isIntegralOrEnumerationType())) {
       SmallString<10> StrBuffer;
-      for (unsigned I = 0; I < Value.getArraySize(); ++I) {
+      for (unsigned I = 0; I < Value.getArrayInitializedElts(); ++I) {
```

Comment:
Note for reviewers: this particular change was done to avoid assertion failure within `APInt` on the following test case:

https://github.com/intel/llvm/blob/c02a98cc705ce056268789382ff9f9afe74c747e/clang/test/SemaSYCL/attr-add-ir-attributes.cpp#L58